h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,239 @@
+import hex.schemas.DRFV3;
+import hex.schemas.DeepLearningV3;
+import hex.schemas.GBMV3;
+import hex.schemas.GLMV3;
+import hex.tree.drf.DRFModel;
+import hex.tree.gbm.GBMModel;
+import hex.tree.xgboost.XGBoostModel;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.api.schemas3.ModelParametersSchemaV3;
+import water.fvec.Frame;
+import water.util.TwoDimTable;
+import static hex.Infogram.InfogramModel.InfogramParameters;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.DoubleStream;
+
+public class InfogramUtils {
+
+  /**
+   * This method will take the columns of _parms.train().  It will then remove the response, any columns in 
+   * _parms._sensitive_attributes from the columns of _parms.train(), weights_column, offset_column.  Then, the 
+   * columns that are left are the columns that are eligible to get their InfoGram.
+   *
+   * @param parms
+   * @return
+   */
+  public static String[] extractPredictors(InfogramParameters parms) {
+    Frame train = parms.train();
+    List<String> colNames = new ArrayList<>(Arrays.asList(parms.train().names()));
+    List<String> excludeCols = new ArrayList<>(Arrays.asList(parms._response_column));
+    if (!(parms._protected_columns == null))
+      excludeCols.addAll(Arrays.asList(parms._protected_columns));  // remove sensitive attributes
+    if (parms._weights_column != null)
+      excludeCols.add(parms._weights_column);  // remove weight
+    if (parms._offset_column != null)
+      excludeCols.add(parms._offset_column);  // remove offset
+    if (parms._ignored_columns != null)
+      excludeCols.addAll(Arrays.asList(parms._ignored_columns));
+    colNames.removeAll(excludeCols);  // remove sensitive attributes, response, weight/offset columns
+    excludeCols = new ArrayList<>();  // reset excludeCols
+    for (String oneCol : colNames)
+      if (train.vec(oneCol).isBad() || train.vec(oneCol).isConst()) 
+        excludeCols.add(oneCol);
+      
+    colNames.removeAll(excludeCols); // remove constant and bad columns
+    return colNames.toArray(new String[colNames.size()]);
+  }
+
+  /**
+   * Method to run infogram model once in order to get the variable importance of the topK predictors
+   * @param parms
+   * @param trainFrame
+   * @param eligiblePredictors
+   * @return
+   */
+  public static String[] extractTopKPredictors(InfogramParameters parms, Frame trainFrame,
+                                               String[] eligiblePredictors, List<Key<Frame>> generatedFrameKeys) {
+    if (parms._top_n_features >= eligiblePredictors.length) return eligiblePredictors;
+    Frame topTrain = extractTrainingFrame(parms, eligiblePredictors, 1, trainFrame);
+    generatedFrameKeys.add(topTrain._key);
+    parms._infogram_algorithm_parameters._train = topTrain._key;
+    ModelBuilder builder = ModelBuilder.make(parms._infogram_algorithm_parameters);
+    Model builtModel = (Model) builder.trainModel().get();
```

Comment:
no risk of leaking keys here ?
- `topTrain` created and added to DKV.
- model building can fail.
- useless `topTrain` remains.

Reply:
I have struggled with making sure there is no leaked keys.  Yes, this code has no leaked keys in my tests.

Reply:
even if one of 
```java
    ModelBuilder builder = ModelBuilder.make(parms._infogram_algorithm_parameters);
    Model builtModel = (Model) builder.trainModel().get();
```
fails? This can easily happen.

Reply:
basically just suggesting that `topTrain` may need to be removed in a `catch` block.

Reply:
Moved to try block